pkg/client/unversioned/clientcmd/loader.go

```diff
@@ -148,6 +164,36 @@ func NewDefaultClientConfigLoadingRules() *ClientConfigLoadingRules {
 	}
 }
 
+func loadFileOrDir(fileOrDir string) ([]string, error) {
+	stat, err := os.Stat(fileOrDir)
+	if err != nil {
+		return []string{}, err
+	}
+	if stat.IsDir() {
+		return loadConfigDir(fileOrDir)
+	}
+	return []string{fileOrDir}, nil
```

Comment:
I'd like to see an explicit sort done here.  One that does a proper numeric sort so people can predict the order files will load in. (`12_foo` after `1_foo`)